core/src/engine/task/task_context.cpp

```diff
@@ -98,12 +98,13 @@ auto* const kFinishedDetachedToken =
 
 TaskContext::TaskContext(TaskProcessor& task_processor,
                          Task::Importance importance, Task::WaitMode wait_type,
-                         Deadline deadline, TaskPayload&& payload)
+                         Deadline deadline,
+                         utils::impl::WrappedCallBase* payload)
     : magic_(kMagic),
       task_processor_(task_processor),
       task_counter_token_(task_processor_.GetTaskCounter()),
       is_critical_(importance == Task::Importance::kCritical),
-      payload_(std::move(payload)),
+      payload_(payload),
```

Comment:
Either UASSERT that the paylpad in not null, or even better: accept a reference here